### PR TITLE
DatePicker: Restore focus after cycling out of the callout

### PR DESCRIPTION
**Pull request by c-w**
#### Pull request checklist

- [x] Addresses an existing issue: #1266
- [x] Include a change request file if publishing
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

After focus leaves the date picker, it skips everything else on the page, making it impossible to tab through the whole page. This pull request fixes this behavior by restoring focus to the date picker's text input after the picker is dismissed.

#### Focus areas to test

Test steps (verified manually on DatePicker.Input.Example: [screencast](https://www.screencast.com/t/yPgjYiT8W)):

1) Tab through the picker until the focus exists. Focus should be restored to the picker's input field.
2) Continue tabbing through the document below the picker.

(Split out from #1451)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -158,6 +158,12 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
     if (ev.which === KeyCodes.enter) {
       ev.preventDefault();
       this._onGotoToday();
+    } else if (ev.which === KeyCodes.tab && !ev.shiftKey) {
```
Comment (MLoughry) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: You should add similar logic for `shift+tab` on the first tab stop of the Calendar

Comment (c-w) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Done in 61b91e3

**Comment by c-w** [comment]
Quick ping to @MLoughry to hit the merge button before the master advances again :)

**Comment by micahgodbolt** [comment]
@c-w do you not have merge access? We typically allow the PR creator to merge the PR once it has been approved. If not, I'll do it for you now.

**Pull request merged**